mirage/config.ml

```diff
@@ -155,24 +82,35 @@ let internal_stack =
     (generic_stackv4v6 ~group:"internal" internal_netif)
     external_stack
 
+(* FIXME: should be exposed in mirage? *)
+let runtime_network_key ~pos ~name fmt =
+  Fmt.kstr
+    (Runtime_arg.create ~name ~pos
+       ~packages:[ package "mirage-runtime" ~sublibs:[ "network" ] ])
+    ("Mirage_runtime_network." ^^ fmt)
+
+let ip_address = Arg.conv (Ipaddr.of_string, Ipaddr.pp)
+let required = Key.Arg.required
+let opt = Key.Arg.opt
+
 let mirage_monitoring =
   let ip_key =
     let ip =
       let doc =
-        Key.Arg.info ~doc:"IP of InfluxDB server to transmit metrics to"
+        Arg.info ~doc:"IP of InfluxDB server to transmit metrics to"
           [ "metrics-ip" ]
       in
-      Key.(create "metrics-ip" Arg.(required ip_address doc))
+      Key.create "metrics-ip" (required ip_address doc)
```

Comment:
@hannesm: what's the intended use of this key? This is now failing at configuration time if you don't provide a `--metrics-ip`, which I suspect is not what is expected. Should we turn those keys to runtime keys?

Reply:
I guess that is the right path to take. The "intended use" is described above, "IP of InfluxDB server to transmit metrics to" -- so yes this should be a runtime key.

Reply:
https://github.com/ocaml/opam-repository/pull/25649